build_images.sh

```diff
@@ -79,6 +83,21 @@ function get_authenticated_docker_command() {
     echo "$docker_authd"
 }
 
+function update_previous_build_sha() {
+    check_vars GIT_COMMIT GIT_BRANCH || return 1
+
+    echo "$GIT_COMMIT" > $PREVIOUS_BUILD_SHA_FILE
```

Comment:
nit: This is usually saner written as

```suggestion
    echo "$GIT_COMMIT" > "$PREVIOUS_BUILD_SHA_FILE"
```